### PR TITLE
[ET-927] ET MA BE Part 2: Implement Orders API and integrate with Attendees API work

### PR DESCRIPTION
**Pull request by sc0ttkclark**
[ET-927]

This work implements the Orders API necessary for Event Tickets Plus to leverage for WooCommerce and EDD Attendees API.

[ET-927]: https://theeventscalendar.atlassian.net/browse/ET-927

### CODE REVIEW

File: src/Tribe/Attendee_Repository.php [hunk 1]
```diff
@@ -1158,4 +1094,71 @@ public function trigger_update_actions( $attendee_data ) {
...
+		/**
+		 * Allow filtering the order data being used to create an order for the attendee.
+		 *
+		 * @since TBD
+		 *
+		 * @param array                         $order_args    List of order data to be saved.
+		 * @param array                         $attendee_data List of additional attendee data.
+		 * @param Tribe__Tickets__Ticket_Object $ticket        The ticket object or null if not relying on it.
+		 */
+		$order_args = apply_filters( 'tribe_tickets_attendee_repository_create_order_for_attendee_order_args', $order_args, $attendee_data, $ticket );
```
Comment (juanfra) [inline src/Tribe/Attendee_Repository.php:1144]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Admin/Manager/Service_Provider.php [hunk 2]
```diff
@@ -67,19 +67,18 @@ public function ajax_handle_admin_manager() {
...
 		 * @since TBD
 		 *
-		 * @param array $vars The request vars.
-		 *
-		 * @return mixed The response.
+		 * @param string|\WP_Error $render_response The render response HTML content or WP_Error with list of errors.
+		 * @param array            $vars            The request variables.
 		 */
-		$render_response = apply_filters( 'tribe_tickets_admin_manager_request', $vars );
+		$render_response = apply_filters( 'tribe_tickets_admin_manager_request', '', $vars );
```
Comment (juanfra) [inline src/Tribe/Admin/Manager/Service_Provider.php:73]: Should we start prefixing without `tribe_` already? `tec_`?

File: src/Tribe/Repositories/Order.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		$status_mgr = tribe( 'tickets.status' );
+
+		/**
+		 * Allow filtering the list of all order statuses supported by the Orders repository.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $statuses List of all order statuses.
+		 */
+		$statuses = apply_filters( 'tribe_tickets_repositories_order_statuses', [] );
```
Comment (juanfra) [inline src/Tribe/Repositories/Order.php:107]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Repositories/Order.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		self::$order_statuses = $statuses;
+
+		/**
+		 * Allow filtering the list of public order statuses supported by the Orders repository.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $public_order_statuses List of public order statuses.
+		 */
+		self::$public_order_statuses = apply_filters( 'tribe_tickets_repositories_order_public_statuses', [] );
```
Comment (juanfra) [inline src/Tribe/Repositories/Order.php:125]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Repositories/Order.php [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		$private_order_statuses = array_diff( self::$order_statuses, self::$public_order_statuses );
+
+		/**
+		 * Allow filtering the list of private order statuses supported by the Orders repository.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $private_order_statuses List of private order statuses.
+		 */
+		self::$private_order_statuses = apply_filters( 'tribe_tickets_repositories_order_private_statuses', $private_order_statuses );
```
Comment (juanfra) [inline src/Tribe/Repositories/Order.php:137]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Tickets.php [hunk 6]
```diff
@@ -3907,11 +3909,107 @@ public function update_attendee( $attendee, $attendee_data ) {
...
+
+			/**
+			 * Allow filtering whether to enable user lookups by Attendee Email.
+			 *
+			 * @since TBD
+			 *
+			 * @param bool  $lookup_user_from_email Whether to lookup the User using the Attendee Email if User ID is not set.
+			 * @param array $args                   The arguments being set for this attendee.
+			 */
+			$lookup_user_from_email = (bool) apply_filters( 'tribe_tickets_attendee_lookup_user_from_email', $lookup_user_from_email, $args );
```
Comment (juanfra) [inline src/Tribe/Tickets.php:3948]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Tickets.php [hunk 7]
```diff
@@ -3907,11 +3909,107 @@ public function update_attendee( $attendee, $attendee_data ) {
...
+
+			/**
+			 * Allow filtering whether to enable creating users using the Attendee Email.
+			 *
+			 * @since TBD
+			 *
+			 * @param bool  $create_user_from_email Whether to create the User using the Attendee Email if User ID is not set.
+			 * @param array $args                   The arguments being set for this attendee.
+			 */
+			$create_user_from_email = (bool) apply_filters( 'tribe_tickets_attendee_create_user_from_email', $create_user_from_email, $args );
```
Comment (juanfra) [inline src/Tribe/Tickets.php:3967]: Should we start prefixing without `tribe_` already?

File: src/Tribe/Tickets.php [hunk 8]
```diff
@@ -3907,11 +3909,107 @@ public function update_attendee( $attendee, $attendee_data ) {
...
+
+			/**
+			 * Allow filtering whether to send the new user information email to the new user.
+			 *
+			 * @since TBD
+			 *
+			 * @param bool  $send_new_user_info Whether to send the new user information email to the new user.
+			 * @param array $args               The arguments being set for this attendee.
+			 */
+			$send_new_user_info = (bool) apply_filters( 'tribe_tickets_attendee_create_user_from_email_send_new_user_info', $send_new_user_info, $args );
```
Comment (juanfra) [inline src/Tribe/Tickets.php:4006]: Should we start prefixing without `tribe_` already?

File: src/template-tags/tickets.php [hunk 9]
```diff
@@ -1296,6 +1296,35 @@ function tribe_attendees( $repository = 'default' ) {
...
+
+		/**
+		 * Filters the map relating Order repository slugs to service container bindings.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $map        A map in the shape [ <repository_slug> => <service_name> ].
+		 * @param string $repository The currently requested implementation.
+		 */
+		$map = apply_filters( 'tribe_tickets_repositories_order_map', $map, $repository );
```
Comment (juanfra) [inline src/template-tags/tickets.php:1322]: Should we start prefixing without `tribe_` already?

**Review by juanfra — APPROVED** [review]
Looking great! Just questions about prefixing for newly introduced hooks

File: src/Tribe/Attendee_Repository.php (hunk 1 above)
Comment (bordoni) [reply src/Tribe/Attendee_Repository.php:1144]: Not just yet. We need to make a decision on this as a group. Lets keep as `tribe_` for now.

**Comment by sc0ttkclark** [comment]
Holding on merge of https://github.com/the-events-calendar/event-tickets/pull/1909

**Review by juanfra — APPROVED** [review]

**Pull request merged**